x-pack/plugins/canvas/public/components/workpad_header/fullscreen_control/fullscreen_control.tsx

```diff
@@ -44,14 +44,26 @@ export class FullscreenControl extends React.PureComponent<Props> {
     setFullscreen(!isFullscreen);
   };
 
+  toggleAutoplay = () => {
+    this.props.toggleAutoplay();
+  };
+
+  nextPage = () => {
+    this.props.nextPage();
+  };
+
+  previousPage = () => {
+    this.props.previousPage();
+  };
```

Comment:
these are all pretty much unnecessary.  Just destructure props and pass through?

```
const { toggleAutoplay, nextPage, previousPage } = this.props;
```

Reply:
You would think that, but I spent a day trying to figure out why you can't.  

The `ReactShortcuts` binds it's handlers when it mounts, and then doesn't do any rebinding if it's props changes.  So if  you bind a shortcut as `nextPage: this.props.nextPage`, it's locked into that specific function.  The nextPage function changes as the active page changes, so that was causing problems. The class functions solve it because they are bound to the components `this` context, so they will always use the most up to date props.

Reply:
OMG, horrid.  Let's document that in comments, please.